### PR TITLE
Dev altitude service

### PR DESCRIPTION
**Pull request by EnBakir**
Provides altitude from backend

### CODE REVIEW

File: src/services/provider/altitude.provider.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { $injector } from '../../injection';
+
+export const loadBvvAltitude = async (easting, northing) => {
```
Comment (taulinger) [inline src/services/provider/altitude.provider.js:3]: We have the type `coordinate` for that purpose

File: src/services/provider/altitude.provider.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+	const result = await httpService.fetch(`${url}/${easting}/${northing} `, {
+		mode: 'cors'
+	});
+
+	if (result.ok) {
+		return await result.json();
+	}
+	throw new Error('Altitude could not be retrieved');
+};
```
Comment (taulinger) [inline src/services/provider/altitude.provider.js:18]: Test for this provider?

File: src/services/AltitudeService.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export class AltitudeService {
+
+	constructor(altitudeProvider = loadBvvAltitude) {
+		this._altitudeProvider = altitudeProvider;
+	} 
+
+	getAltitudeProvider() {
+		return this._altitudeProvider;
+	}
```
Comment (taulinger) [inline src/services/AltitudeService.js:11]: The service should provide a method `getAltitude(coordinate)` which uses the provider internally. No need to expose the provider...

File: test/service/AltitudeService.test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+		return new AltitudeService(provider);
+	}; 
+
+	describe('init', () => {
+
+		it('initializes the service', async () => {
+			const instanceUnderTest = setup();
+			expect(instanceUnderTest._altitudeProvider).toBeDefined();
+			expect(instanceUnderTest._altitudeProvider).toEqual(loadBvvAltitude);
+		});
```
Comment (taulinger) [inline test/service/AltitudeService.test.js:16]: We also need tests for bad responses from the provider...

File: src/services/AltitudeService.js [hunk 5]
```diff
@@ -4,10 +4,34 @@ export class AltitudeService {
...
 	} 
 
-	getAltitudeProvider() {
-		return this._altitudeProvider;
-	} 
+	/**
+     * 
+     * @param {Coordinate} coordinate3857 
+     */
+	async getAltitude(coordinate3857) {
```
Comment (taulinger) [inline src/services/AltitudeService.js:14]: Should return just a double value

File: src/services/provider/altitude.provider.js [hunk 6]
```diff
@@ -1,13 +1,17 @@
 import { $injector } from '../../injection';
 
-export const loadBvvAltitude = async (easting, northing) => {
+/**
+ * 
+ * @param {Coordinate} coordinate3857
+ */
+export const loadBvvAltitude = async (coordinate3857) => {
```
Comment (taulinger) [inline src/services/provider/altitude.provider.js:7]: Should also return just a number, no need to expose the backend payload

File: src/services/AltitudeService.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+		if (coordinate3857) {
+			try {
+				this._altitude = await this._altitudeProvider(coordinate3857);
+				return this._altitude;
+			}
+			catch (e) {
+				return Promise.reject('AltitudeService could not be initialized: ' + e.message);
```
Comment (taulinger) [inline src/services/AltitudeService.js:21]: Correct error message?

File: test/service/AltitudeService.test.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+		it('initializes the service with default provider', async () => {
+			const instanceUnderTest = new AltitudeService();
+			expect(instanceUnderTest._altitudeProvider).toEqual(loadBvvAltitude);
+		});
```
Comment (taulinger) [inline test/service/AltitudeService.test.js:21]: Both tests use the default provider.
Shouldn't be the first test against a custom provider?

File: src/modules/map/components/olMap/handler/contextMenu/OlMapContextMenuContent.js [hunk 9]
```diff
@@ -13,17 +13,36 @@ export class OlMapContextMenuContent extends BaElement {
...
 		this._coordinate = coordinateInMapSrid;
+		this._getAltitude();
+	}
+
+	/**
+	 * @private
+	 */
+	async _getAltitude() {
+		try {
+			this._altitude = await this._altitudeService.getAltitude(this._coordinate);
```
Comment (taulinger) [inline src/modules/map/components/olMap/handler/contextMenu/OlMapContextMenuContent.js:39]: We should also display a unit (m)  😉

**Review by thiloSchlemmer — APPROVED** [review]

**Review by taulinger — COMMENTED** [review]
Refactoring is done, I think we can change this branch to a PR

**Comment by taulinger** [comment]
I would suggest @thiloSchlemmer for the review

**Review by thiloSchlemmer — APPROVED** [review]

**Review by taulinger — APPROVED** [review]

**Pull request merged**